kolibri/logger/management/commands/userdata_utils.py

```diff
@@ -0,0 +1,327 @@
+from kolibri.auth.constants.role_kinds import ADMIN
+from kolibri.auth.filters import HierarchyRelationsFilter
+from kolibri.auth.models import Classroom, Facility, FacilityUser
+from kolibri.content.models import ContentNode
+from kolibri.core.device.models import DeviceSettings, DevicePermissions
+from kolibri.logger.models import AttemptLog, ContentSessionLog, ContentSummaryLog, MasteryLog
+from le_utils.constants import content_kinds
+
+
+def provision_device():
+    print('Provisioning device. Onboarding will be skipped after starting server.')
+    device_settings, created = DeviceSettings.objects.get_or_create()
+    device_settings.is_provisioned = True
+    device_settings.language_id = 'en'
+    device_settings.save()
+
+
+def add_superuser_to_facility(facility):
+    print('Creating superuser "admin" with password "admin" at facility {facility}.'.format(facility=facility.name))
+    superuser = FacilityUser.objects.create(
+        facility=facility,
+        full_name='Ad Min',
+        username='admin',
+    )
+    superuser.set_password('admin')
+    superuser.save()
+    DevicePermissions.objects.create(user=superuser, is_superuser=True)
+    facility.add_role(superuser, ADMIN)
+
+
+def get_or_create_facilities(**options):
+    n_facilities = options['n_facilities']
+    n_on_device = Facility.objects.all().count()
+    n_to_create = n_facilities - n_on_device
+    if n_to_create > 0:
+        print('Generating {n} facility object(s)'.format(n=n_to_create))
+        for i in range(0, n_to_create):
+            Facility.objects.create(name='Test Facility {i}'.format(i=i + 1))
+    return Facility.objects.all()[0:n_facilities]
+
+
+def get_or_create_classrooms(**options):
+    n_classes = options['n_classes']
+    facility = options['facility']
+    n_on_device = Classroom.objects.filter(parent=facility).count()
+    n_to_create = n_classes - n_on_device
+    if n_to_create > 0:
+        print('Generating {n} classroom object(s) for facility: {name}'.format(
+            n=n_to_create,
+            name=facility.name,
+        ))
+        for i in range(0, n_to_create):
+            Classroom.objects.create(
+                parent=facility,
+                name='Classroom {i}{a}'.format(i=i + 1, a=random.choice('ABCD'))
+            )
+    return Classroom.objects.filter(parent=facility)[0:n_classes]
+
+
+def get_or_create_classroom_users(**options):
+    classroom = options['classroom']
+    n_users = options['n_users']
+    user_data = options['user_data']
+    facility = options['facility']
+
+    # The headers in the user_data.csv file that we use to generate user Full Names
+    # Note, we randomly pick from these to give deliberately varied (and sometimes idiosyncratic)
+    # Full names - because we should never assume that users have names like us
+    user_data_name_fields = ["GivenName", "MiddleInitial", "Surname"]
+
+    n_on_device = HierarchyRelationsFilter(FacilityUser.objects.all()).filter_by_hierarchy(
```

Comment:
This is a slightly misleading variable name, as it is the number of users in the classroom, not on the device in total.